phoenix-core/src/main/java/org/apache/phoenix/compile/ProjectionCompiler.java

```diff
@@ -641,19 +702,23 @@ public <T> T accept(ExpressionVisitor<T> visitor) {
             return null;
         }
     }
-    private static void serailizeArrayIndexInformationAndSetInScan(StatementContext context, List<Expression> arrayKVFuncs,
-            List<KeyValueColumnExpression> arrayKVRefs) {
+
+    private static void serializeServerParsedExpressionInformationAndSetInScan(
+            StatementContext context, String serverParsedExpressionAttribute,
+            List<Expression> serverParsedKVFuncs,
+            List<KeyValueColumnExpression> serverParsedKVRefs) {
         ByteArrayOutputStream stream = new ByteArrayOutputStream();
         try {
             DataOutputStream output = new DataOutputStream(stream);
-            // Write the arrayKVRef size followed by the keyvalues that needs to be of type arrayindex function
-            WritableUtils.writeVInt(output, arrayKVRefs.size());
-            for (Expression expression : arrayKVRefs) {
+            // Write the arrayKVRef size followed by the keyvalues that needs to be of
+            // type arrayindex+json function
```

Comment:
Isn't this comment should be arrayIndex **OR** json. We serialize all of these types independently

Reply:
right, its based on the attribute. will change for clarity